be/src/vec/functions/function_ip.h

```diff
@@ -1318,9 +1321,10 @@ class FunctionCutIPv6 : public IFunction {
                                 get_name());
             }
 
-            UInt8 bytes_to_cut_count = is_ipv4_mapped(address) ? bytes_to_cut_for_ipv4_count
-                                                               : bytes_to_cut_for_ipv6_count;
-            cut_address(address, pos, bytes_to_cut_count);
+            UInt8 bytes_to_cut_count = is_ipv4_mapped(ipv6_address_data)
+                                               ? bytes_to_cut_for_ipv4_count
+                                               : bytes_to_cut_for_ipv6_count;
+            cut_address(ipv6_address_data, pos, bytes_to_cut_count);
```

Comment:
Add detail comment for the code. And consider we pass reference or value replace by ptr